samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -743,7 +745,9 @@ int location_shell(const struct shell *shell, size_t argc, char **argv)
 		for (uint8_t i = 0; i < method_count; i++) {
 			if (config.methods[i].method == LOCATION_METHOD_GNSS) {
 				if (gnss_timeout_set) {
-					config.methods[i].gnss.timeout = gnss_timeout;
+					config.methods[i].gnss.timeout =
+						(gnss_timeout == SYS_FOREVER_MS) ?
```

Comment:
Ah, this (exact comparison) could be a problem with floating point numbers. It would be safer to use `SYS_FOREVER_MS` if `gnss_timeout` is negative?

Reply:
It seems to work. I was pretty sure this wouldn't work but it does. We should discuss though how to make this as I'm not sure if I'm just lucky that it works fine.

Reply:
> It would be safer to use `SYS_FOREVER_MS` if `gnss_timeout` is negative?

If we would do this, you couldn't use MOSH to test how Location library works with negative timeout values.

Reply:
Just checked and I thought `SYS_FOREVER_MS` would be `0xFFFFFFFF` but it's actually this (which I think should work with floats as well):
`#define SYS_FOREVER_MS (-1)`

Reply:
Yes, it's -1. I guess in this case it works because you just assign the value and compare it without any arithmetics, but at least I have avoided using == and != with floating point numbers, because they are tricky. At least I'm not sure if that always works. 😄

Reply:
In general, you are correct. But as you say, we are setting a simple value and then comparing against it. So this should work. There shouldn't be any indeterministic behavior related to it as a bit for negative value in the float is set in addition to the bit for indicating value '1'.

Reply:
Floating point numbers are not stored like integers. Even if you do not perform any arithmetic operations, the value may not be exactly what you would expect. Take the following example:
```
#include <stdio.h>

int main(void)
{
	float a;

	a = -1.0;

	if (a == -1.0) {
		printf("a is -1.0\n");
	} else {
		printf("a is not -1.0\n");
	}

	a = -1.1;

	if (a == -1.1) {
		printf("a is -1.1\n");
	} else {
		printf("a is not -1.1\n");
	}

	return 0;
}
```
When executed, the first comparison is true, while the second is not. To further complicate things, changing the `float` to a `double` makes the second comparison also true. It works like this on my PC, it might behave differently on different platforms.

I think it's good practice to avoid things which are known to be error prone.

Reply:
Changed as discussed to check zero value in a different place of the code.